test/test_util.jl

```diff
@@ -20,9 +20,57 @@ macro trixi_testset(name, expr)
         using Test
         using TrixiBase
 
+        # We also include this file again to provide the definition of
+        # the other testing macros. This allows to use `@trixi_testset`
+        # in a nested fashion and also call `@test_nowarn_mod` from
+        # there.
+        include(@__FILE__)
+
         @testset verbose=true $name $expr
         end
 
         nothing
     end
 end
+
+"""
+    @test_nowarn_mod expr
+
+Modified version of `@test_nowarn expr` that prints the content of `stderr` when
+it is not empty and ignores some common info statements printed in Trixi.jl
+uses.
+"""
+macro test_nowarn_mod(expr, additional_ignore_content=String[])
```

Comment:
Can we somehow export this as well when importing TrixiBase from a testing environment? Then we can avoid duplicating this function from Trixi.

Reply:
@ranocha?

Reply:
@efaulhaber Maybe we can move this convo to a separate issue such that we can merge this PR (once you approve) and then start the registration process?

Reply:
We would need to move this macro to the plain source code of the package and make TrixiBase.jl depend on Test.

Reply:
So there is no nice way to do it? Maybe add Test as a soft dependency or so?

Reply:
As @sleode said, I'm fine discussing this further in an issue and merging this soon to start the registration process.

Reply:
I tend to prefer moving this to a TrixiTest.jl package to avoid depending on Test etc. for everything

Reply:
Moved to #9. Your points are absolutely valid @efaulhaber, I just wanted to make sure we can move forward with the current state.